src/commands/project.lisp

```diff
@@ -221,3 +224,85 @@
                                        (incf count))))
       (delete-buffers (list-project-buffers)))
     (message "~a buffers deleted." count)))
+
+
+
+(defvar *projects-history*)
+
+(defun history ()
+  "Return or create the projects' history struct.
+  The history file is saved on (lem-home)/history/projects"
+  (unless (boundp '*projects-history*)
+    (let* ((pathname (merge-pathnames "history/projects" (lem-home)))
+           (history (lem/common/history:make-history :pathname pathname)))
+      (setf *projects-history* history)))
+  *projects-history*)
+
+(defun remember-project (input)
+  "Add this project path to the history file.
+  Return t if the project was added, nil if it already existed."
+  (let* ((history (history))
+         (data (lem/common/history:history-data history))
```

Comment:
How about defining a function that searches through the history to facilitate encapsulation?

Reply:
I addressed my needs in history-add by adding a &key allow-duplicates.